### DISCUSSION TITLE
RTranslator 2.0.3

### DISCUSSION THREAD

**Discussion by niedev** [niedev/RTranslator#48] [General]
**Improvements to the initial download of the AI models:**

- **Added a file integrity check to the initial download**. Now you will be notified if a downloaded file is corrupted and you can restart the download of only that file (previously the corruption was only identified at the end of the download and you had to reset the app and start over).

- **Added an option to fix eventually corrupted file models at the start of the app**. Now, if the error occurs, instead of just suggesting to reinstall the app, you can click on the "fix" button and re-download only the corrupted files.

- **Added the option of sideloading**. People who previously had problems downloading due to poor internet connection to GitHub can now do the initial download wherever they want and then push the files into the app. See the [dedicated guide](https://github.com/niedev/RTranslator/blob/v2.00/Sideloading.md) for more info.

- **Added an eventual low storage space warning in the initial download**

<hr /><em>This discussion was created from the release <a href='https://github.com/niedev/RTranslator/releases/tag/2.0.3'>RTranslator 2.0.3</a>.</em>

**Comment by fishergz** [comment]
Is it possible to add a feature to show the original text recognized by Whisper?
It’s easier to know what is being translated

**Reply by niedev** [reply]
Hi, it's already there, just select the same language for both languages in WalkieTalkie mode and the translation doesn't happen, just transcription.

**Reply by fishergz** [reply]
What I mean is, before WalkieTalkie translates it, let me see if there are any errors in the text recognized by whisper.

**Reply by niedev** [reply]
Oh ok, but what's the utility of this information?

**Reply by fishergz** [reply]
Thank you, because I can’t understand the translated content, so I can confirm whether wisper heard my original words correctly.

**Reply by niedev** [reply]
Ok, I will think about it 👍
